core/src/mender-client.c

```diff
@@ -805,6 +801,48 @@ mender_client_initialization_work_function(void) {
     return ret;
 }
 
+static mender_err_t
+mender_commit_artifact_data(void) {
+
+    cJSON *json_artifact_name = cJSON_GetObjectItemCaseSensitive(mender_client_deployment_data, "artifact_name");
+    if (NULL == json_artifact_name) {
+        mender_log_error("Unable to get artifact name from the deployment data");
+        return MENDER_FAIL;
+    }
+
+#ifdef CONFIG_MENDER_FULL_PARSE_ARTIFACT
+#ifdef CONFIG_MENDER_PROVIDES_DEPENDS
+    /* Get provides from the deployment data */
+    cJSON *provides = cJSON_GetObjectItemCaseSensitive(mender_client_deployment_data, "provides");
+    if (NULL == provides) {
+        mender_log_error("Unable to get new_provides from the deployment data");
+        return MENDER_FAIL;
+    }
+
+    /* Parse provides */
+    mender_key_value_list_t *new_provides = NULL;
+    if (MENDER_OK != mender_utils_string_to_key_value_list(provides->valuestring, &new_provides)) {
+        mender_log_error("Unable to parse provides from the deployment data");
+        return MENDER_FAIL;
+    }
+    /* Replace the stored provides with the new provides */
+    if (MENDER_OK != mender_storage_set_provides(new_provides)) {
+        mender_log_error("Unable to set provides");
+        mender_utils_free_linked_list(new_provides);
+        return MENDER_FAIL;
+    }
+    mender_utils_free_linked_list(new_provides);
+#endif /* CONFIG_MENDER_PROVIDES_DEPENDS */
+#endif /* CONFIG_MENDER_FULL_PARSE_ARTIFACT */
+
+    if (MENDER_OK != mender_storage_set_artifact_name(json_artifact_name->valuestring)) {
+        mender_log_error("Unable to set artifact name");
+        return MENDER_FAIL;
+    }
```

Comment:
I'd do this first to keep the code dealing with `artifact_name` together. But not a biggie.

Reply:
Yeah, I agree :+1: